packages/web3-eth-accounts/src/index.js

```diff
@@ -287,6 +313,93 @@ function _validateTransactionForSigning(tx) {
     return;
 }
 
+function _handleTxType(tx) {
+    let txType = tx.type !== undefined ? utils.toHex(tx.type) : '0x0';
+    // Taken from https://github.com/ethers-io/ethers.js/blob/2a7ce0e72a1e0c9469e10392b0329e75e341cf18/packages/abstract-signer/src.ts/index.ts#L215
+    const hasEip1559 = (tx.maxFeePerGas !== undefined || tx.maxPriorityFeePerGas !== undefined);
+    if (tx.gasPrice !== undefined && (tx.type === '0x2' || hasEip1559))
+        throw Error("eip-1559 transactions don't support gasPrice");
+    if ((tx.type === '0x1' || tx.type === '0x0') && hasEip1559)
+        throw Error("pre-eip-1559 transaction don't support maxFeePerGas/maxPriorityFeePerGas");
+    
+    if (
+        hasEip1559 ||
+        (
+            (tx.common && tx.common.hardfork && tx.common.hardfork.toLowerCase() === 'london') ||
+            (tx.hardfork && tx.hardfork.toLowerCase() === 'london')
+        )
+    ) {
+        txType = '0x2';
+    } else if (
+        tx.accessList ||
+        (
+            (tx.common && tx.common.hardfork && tx.common.hardfork.toLowerCase() === 'berlin') ||
+            (tx.hardfork && tx.hardfork.toLowerCase() === 'berlin')
+        )
+    ) {
+        txType = '0x1';
+    }
+    
+    return txType
+}
+
+function _handleTxPricing(_this, tx) {
+    return new Promise((resolve, reject) => {
+        try {
+            if (
+                (
+                    tx.type === undefined ||
+                    tx.type === '0x0' ||
+                    tx.type === '0x1'
+                ) && tx.gasPrice !== undefined)
+            {
+                // Legacy transaction, return provided gasPrice
+                resolve({ gasPrice: tx.gasPrice })
+            } else {
+                Promise.all([
+                    _this._ethereumCall.getBlockByNumber(),
+                    _this._ethereumCall.getGasPrice()
+                ]).then(responses => {
+                    const [block, gasPrice] = responses;
+                    if (
+                        (tx.type !== '0x0' && 
+                        tx.type !== '0x1') &&
+                        block && block.baseFeePerGas
+                    ) {
+                        // The network supports EIP-1559
+    
+                        // Taken from https://github.com/ethers-io/ethers.js/blob/ba6854bdd5a912fe873d5da494cb5c62c190adde/packages/abstract-provider/src.ts/index.ts#L230
+                        let maxPriorityFeePerGas, maxFeePerGas;
+    
+                        if (tx.gasPrice) {
+                            // Using legacy gasPrice property on an eip-1559 network,
+                            // so use gasPrice as both fee properties
+                            maxPriorityFeePerGas = tx.gasPrice;
+                            maxFeePerGas = tx.gasPrice;
+                            delete tx.gasPrice;
+                        } else {
+                            maxPriorityFeePerGas = tx.maxPriorityFeePerGas || '0x3B9ACA00'; // 1 Gwei
```

Comment:
Its not part of the EIP ( specs )  to must have default 'priority fee per gas' 1 Gwei. Its only suggested to set this default.

so I suggest It should always be coming from web3 user only, or if we are setting default then must notify / inform user as well.

Reply:
This was something that was suggested on a call with wallet/library implementers, but we should make a note in the docs

Reply:
We'll add `maxPriorityFeePerGas` being default to `1 Gwei` to the docs